spring-integration-core/src/main/java/org/springframework/integration/mapping/AbstractHeaderMapper.java

```diff
@@ -423,6 +454,55 @@ public boolean matchHeader(String headerName) {
 			return false;
 		}
 
+		@Override
+		public boolean isNegated() {
+			return false;
+		}
+
+	}
+
+	/**
+	 * A pattern-based {@link HeaderMatcher} that matches if the specified
+	 * header matches the specified simple pattern.
+	 * @see org.springframework.util.PatternMatchUtils#simpleMatch(String, String)
+	 * @since 4.3
+	 */
+	protected static class SinglePatternBasedHeaderMatcher implements HeaderMatcher {
+
+		private static final Log logger = LogFactory.getLog(HeaderMatcher.class);
```

Comment:
I see all classes here use the same logger. 
Can't we extract only one instance for all of them in the outer class to use from here?

Maybe that's minor, but I wonder if we can minimize the count of objects in the memory :smile: